agent/rpc_from.go

```diff
@@ -147,57 +149,61 @@ func (a *orbAgent) handleAgentReset(payload fleet.AgentResetRPCPayload) {
 }
 
 func (a *orbAgent) handleRPCFromCore(client mqtt.Client, message mqtt.Message) {
+	go func() {
```

Comment:
why is this wrapped in a go function now? i don't see how concurrency helps here

Reply:
This is actually a good practice from paho's documentation, when we use the option.OrderMatters(true) which is also the default, it says that that the message handling needs to be a go function. With the tests I did, this closes the connection. 
Here
```go
/ SetOrderMatters will set the message routing to guarantee order within
// each QoS level. By default, this value is true. If set to false (recommended),
// this flag indicates that messages can be delivered asynchronously
// from the client to the application and possibly arrive out of order.
// Specifically, the message handler is called in its own go routine.
// Note that setting this to true does not guarantee in-order delivery
// (this is subject to broker settings like "max_inflight_messages=1" in mosquitto)
// and if true then handlers must not block.
```

In the last line of this it says "and if true then handlers must not block.

https://github.com/eclipse/paho.mqtt.golang/blob/bcb58ee4a1782dd29e40cc37230a1f8a31df1376/options.go#L229-L240

Reply:
Hm ok wouldn't it be better to wrap it in a go function here, rather than change handleRPCFromCore:
https://github.com/ns1labs/orb/blob/1bdabbd43c05c6e3f93a75ea32ab694480d318ec/agent/comms.go#L47

Reply:
hm I am used to that way, on the function is better because it can be tracked better, and it allow us to control things like context better if we ever add context to those calls.